src/Components/Admins/Form/index.js

```diff
@@ -0,0 +1,259 @@
+        password: data.data.password
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  const createAdmin = async () => {
+    try {
+      console.log('formChange', formChange);
+      const createdAdmin = await fetch(`${process.env.REACT_APP_API_URL}/api/admins`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(formChange)
+      });
+
+      const createdAdminsData = await createdAdmin.json();
+      if (createdAdmin.ok) {
+        setFormChange({
+          firstName: '',
+          lastName: '',
+          dni: '',
+          phone: '',
+          email: '',
+          city: '',
+          password: ''
+        });
+        setResponseModal({
+          title: 'Success!',
+          description: createdAdminsData.message,
+          isConfirm: false
+        });
+        setIsOpen(true);
+      } else {
+        setResponseModal({
+          title: 'ERROR!',
+          description: createdAdminsData.message,
+          isConfirm: false
+        });
+        setIsOpen(true);
+      }
+    } catch (error) {
+      setResponseModal({
+        title: 'ERROR!',
+        description: error.message,
+        isConfirm: false
+      });
+      setIsOpen(true);
+    }
+  };
+
+  const updateAdmins = async () => {
+    try {
+      const updatedAdminRes = await fetch(`${process.env.REACT_APP_API_URL}/api/admins/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(formChange)
+      });
+
+      const updatedAdmin = await updatedAdminRes.json();
+      if (updatedAdminRes.ok) {
+        setFormChange({
+          firstName: '',
+          lastName: '',
+          dni: '',
+          phone: '',
+          email: '',
+          city: '',
+          password: ''
+        });
+        setResponseModal({
+          title: 'Success!',
+          description: updatedAdmin.message,
+          isConfirm: false
+        });
+        setIsOpen(true);
+      } else {
+        setResponseModal({
+          title: 'ERROR!',
+          description: updatedAdmin.message,
+          isConfirm: false
+        });
+        setIsOpen(true);
+      }
+    } catch (error) {
+      setResponseModal({
+        title: 'ERROR!',
+        description: error.message,
+        isConfirm: false
+      });
+      setIsOpen(true);
+    }
+  };
+
+  const cancel = () => {
+    history.push('/admins');
+  };
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    if (id) {
+      updateAdmins();
+    } else {
+      createAdmin();
+    }
+  };
+
+  useEffect(() => {
+    if (id) {
+      getAdminById(id);
+    }
+  }, []);
+
+  const onChangeInput = (e) => {
+    setFormChange({
+      ...formChange,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const handleClose = () => {
+    setIsOpen(!isOpen);
+    history.push('/admins');
+  };
+
```

Comment:
We added a validatión, when the form isn't fill correctly the form should remain open after you close the modal
```suggestion
const closeForm = () => {
    if (isActivityCreated) {
      setIsOpen(false);
      history.goBack();
    } else {
      switchIsOpen();
    }
  };
```
we used this function to validate that